pkg/proxy/server.go

```diff
@@ -353,6 +367,40 @@ func (s *server) listenAndServe() {
 			continue
 		}
 
+		parseHeaderForDestination := func() string {
+			// the first request should always contain a CONNECT header field
+			// since we set the transport to forward the traffic to the proxy
+			buf := make([]byte, s.bufferSize)
+			var data []byte
+			if nr1, err := in.Read(buf); err != nil {
+				if err == io.EOF {
```

Comment:
NOTE: During teardown, the connection will be closed and `in.Read` returns EOF. It will panic.

Reply:
Thanks for confirming! I suspected this to be the cause but I will have to check it in the ioCopy function too!

Reply:
I think you can just return error and allow caller to close connection because it's closed by peer.